TODO.md

```diff
@@ -68,3 +68,5 @@
   every read/write range command.
 
 * Gossip should be changed to send only protobufs.
+* persist Gossip and the kv.leaderCache to make sure that restarting
+  a node is harmless.
```

Comment:
Rolling restarts of storage nodes should be infrequent and slow, so I'm not sure how concerned we should be about the network traffic they require to bring each node back into the gossip network (this is just one of the reasons why a restart can be disruptive). 

As for the leader cache, we should probably just add an option to rescan the whole thing (or selected subranges of the keyspace) at startup. If lots of nodes are restarting then a stored leader cache rapidly becomes useless. In all of my bigtable apps I set up a background thread to scan the whole location map every 10 minutes from every client (and doing so was cheap, so it was well worth it to nearly eliminate cache misses in the location map)